src/main/java/lotto/domain/Lotto.java

```diff
@@ -0,0 +1,85 @@
+package lotto.domain;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+public class Lotto {
+    private static final int MINIMUM_NUMBER = 1;
+    private static final int MAXIMUM_NUMBER = 45;
+    private static final int LOTTO_COUNT = 6;
+    private static final String ERROR_LOTTO_COUNT = LOTTO_COUNT + "개의 숫자를 입력해주세요";
+    private static final String ERROR_DUPLICATED_NUMBER = "번호가 중복됩니다!";
+
+    private final List<Ball> lotto = new ArrayList<>();
+
+    public Lotto() {
+        this(selectNumbers());
+    }
+
+    public Lotto(final List<String> numbers) {
+        checkValidNumbers(numbers);
+
+        for (String number : numbers) {
+            this.lotto.add(new Ball(number));
+        }
+    }
+
+    public List<String> getLottoNumbers() {
+        return lotto.stream()
+            .map(Ball::toString)
+            .collect(Collectors.toList());
+    }
+
+    public boolean contains(Ball number) {
+        return lotto.contains(number);
+    }
+
+    public int getMatchingCount(Lotto compareLotto) {
+        return (int) lotto.stream()
+                .filter(compareLotto::contains).count();
```

Comment:
요런 경우에는 가독성을 위해 . 뒤를 한 칸 내려주면 좋아요 😀
```suggestion
                .filter(compareLotto::contains)
                .count();
```